tests/python/test_graph.py

```diff
@@ -21,3 +21,84 @@ def test(pos: ti.types.ndarray(field_dim=1)):
     a = ti.ndarray(ti.i32, shape=(n, ))
     g.run({'pos': a})
     assert (a.to_numpy() == np.ones(4)).all()
+
+
+@test_utils.test(arch=ti.vulkan)
+def test_ndarray_float():
+    n = 4
+
+    @ti.kernel
+    def test(pos: ti.types.ndarray(field_dim=1)):
+        for i in range(n):
+            pos[i] = 2.5
+
+    sym_pos = ti.graph.Arg(ti.graph.ArgKind.NDARRAY, 'pos', ti.f32)
+    g_init = ti.graph.GraphBuilder()
+    g_init.dispatch(test, sym_pos)
+    g = g_init.compile()
+
+    a = ti.ndarray(ti.f32, shape=(n, ))
+    g.run({'pos': a})
+    assert (a.to_numpy() == (np.ones(4) * 2.5)).all()
+
+
+def build_graph(arr_type, N, dtype):
+    @ti.kernel
+    def vector_sum(mat: ti.types.vector(N, dtype),
+                   res: ti.types.ndarray(field_dim=1)):
+        res[0] = mat.sum() + mat[2]
+
+    @ti.kernel
+    def matrix_sum(mat: ti.types.matrix(N, 2, dtype),
+                   res: ti.types.ndarray(field_dim=1)):
+        res[0] = mat.sum()
+
+    sym_A = ti.graph.Arg(ti.graph.ArgKind.MATRIX, 'mat', dtype)
```

Comment:
@turbo0628 Matrix argkind can have element shape, I think we can do this flattening all in  python without hacking in C++?

Reply:
The flattening is all in Python, the MATRIX is actually a scalar. Should I remove this?

Reply:
I just think it a bit weird to expose `ArgKind.SCALAR` for Matrix

Reply:
@turbo0628 There's a slight difference, I'd expect here to be `ti.graph.Arg(ti.graph.ArgKind.Matrix, 'mat', ti.types,matrix(N, 2, dtype))`? and it's flattened in python, so what C++ get is actually `N*2` scalar args.

Reply:
Make sense, will make the changes later today